src/auth/auth.controller.ts

```diff
@@ -52,6 +52,12 @@ export class AuthController {
   private async login(req: LoginRequest): Promise<LoginResponse> {
     let user: User;
     try {
+      // for request with content-type application/x-www-form-urlencoded
+      // body will be in format {'json data': ''}
+      // fastify-formbody still does not parse correctly
+      if (!req.user) {
+        req = JSON.parse(Object.keys(req)[0]);
+      }
```

Comment:
Use `addContentTypeParser` to set expected parser for `application/x-www-form-urlencoded`

Reply:
It does not work with Swagger

```ts
const app: NestFastifyApplication = await NestFactory.create(
    AppModule,
    new FastifyAdapter(server),
    {
      bodyParser: false
    }
  );
  await app.init();
```
```
Content type parser 'application/x-www-form-urlencoded' already present
```

Reply:
Oh, fastify or nestjs set it by default.  All the more... If fastify-formbody does not parse body as expected, then you do something wrong.